internal/httpclient/httpclient.go

```diff
@@ -7,23 +7,18 @@ import (
 	"fmt"
 	"net/http"
 
-	"github.com/hashicorp/go-cleanhttp"
+	"github.com/hashicorp/go-retryablehttp"
 	"github.com/hashicorp/hc-install/version"
 )
 
-// NewHTTPClient provides a pre-configured http.Client
+// NewHTTPClient provides a pre-configured retryablehttp.Client
 // e.g. with relevant User-Agent header
-func NewHTTPClient() *http.Client {
-	client := cleanhttp.DefaultClient()
-
-	userAgent := fmt.Sprintf("hc-install/%s", version.Version())
-
-	cli := cleanhttp.DefaultPooledClient()
-	cli.Transport = &userAgentRoundTripper{
-		userAgent: userAgent,
-		inner:     cli.Transport,
+func NewHTTPClient() *retryablehttp.Client {
```

Comment:
If we use the `.StandardClient()` helper method here then we should not need to change any references to `NewHTTPClient()` as the interface remains the same.